packages/rest/test/acceptance/file-upload/file-upload.acceptance.ts

```diff
@@ -0,0 +1,159 @@
+        filename: 'cert.pem',
+        contentType: 'multipart/form-data',
+      })
+      .expect(200);
+    expect(res.body.files[0]).containEql({
+      fieldname: 'certFile',
+      originalname: 'cert.pem',
+      mimetype: 'multipart/form-data',
+    });
+  });
+
+  function givenBodyParamController() {
+    const spec = anOpenApiSpec()
+      .withOperation('post', '/show-body', {
+        'x-operation-name': 'showBody',
+        requestBody: <RequestBodyObject>{
+          description: 'multipart/form-data value.',
+          required: true,
+          content: {
+            'multipart/form-data': {
+              // Skip body parsing
+              'x-parser': 'stream',
+              schema: {type: 'object'},
+            },
+          },
+        },
+        responses: {
+          200: {
+            content: {
+              'application/json': {
+                schema: {
+                  type: 'object',
+                },
+              },
+            },
+            description: '',
+          },
+        },
+      })
+      .build();
+
+    class RouteParamController {
+      async showBody(
+        request: Request,
+        @inject(RestBindings.Http.RESPONSE) response: Response,
+      ): Promise<Object> {
+        const storage = multer.memoryStorage();
+        const upload = multer({storage});
+        return new Promise<object>((resolve, reject) => {
+          upload.any()(request, response, err => {
+            if (err) reject(err);
+            else {
+              resolve({
+                files: request.files,
+                // tslint:disable-next-line:no-any
+                fields: (request as any).fields,
+              });
+            }
+          });
+        });
+      }
+    }
+
+    givenControllerClass(RouteParamController, spec);
+  }
+
+  function givenControllerClass(
+    // tslint:disable-next-line:no-any
+    ctor: new (...args: any[]) => Object,
+    spec: ControllerSpec,
+  ) {
+    handler.registerController(spec, ctor);
+  }
+
+  let rootContext: Context;
+  let handler: HttpHandler;
+  function givenHandler() {
+    rootContext = new Context();
+    rootContext.bind(SequenceActions.FIND_ROUTE).toProvider(FindRouteProvider);
+    rootContext
+      .bind(SequenceActions.PARSE_PARAMS)
+      .toProvider(ParseParamsProvider);
+    rootContext
+      .bind(SequenceActions.INVOKE_METHOD)
+      .toProvider(InvokeMethodProvider);
+    rootContext
+      .bind(SequenceActions.LOG_ERROR)
+      .to(createUnexpectedHttpErrorLogger());
+    rootContext.bind(SequenceActions.SEND).to(writeResultToResponse);
+    rootContext.bind(SequenceActions.REJECT).toProvider(RejectProvider);
+
+    rootContext.bind(RestBindings.SEQUENCE).toClass(DefaultSequence);
+    rootContext
+      .bind(RestBindings.REQUEST_BODY_PARSER)
+      .toClass(RequestBodyParser);
+
+    handler = new HttpHandler(rootContext);
+    rootContext.bind(RestBindings.HANDLER).to(handler);
+  }
+
+  function givenClient() {
```

Comment:
Similarly here. Since this is an acceptance test, if you use `RestApplication` as the root context, then you can use testlab's client factory functions too.

The original setup code was used in integration-level tests where I did not want to use full application originally. Maybe it's time to revisit that decision and simplify our integration-level tests too, but such refactoring/cleanup out of scope of this pull request. However, that does not mean that new test code should be repeating old mistakes and introducing new tech debt. So let's use RestApplication + testlab helpers in the acceptance tests please.